### PR TITLE
Feature/adjusts for optimas

### PR DESCRIPTION
**Pull request by jlnav**
Some tentative changes, _mostly_ adjusts to the APOSMM class so it does internal buffering

### CODE REVIEW

File: libensemble/gen_classes/aposmm.py [hunk 1]
```diff
@@ -28,20 +29,46 @@ def __init__(
                 ("local_min", bool),
                 ("local_pt", bool),
             ]
-            gen_specs["persis_in"] = ["x", "f", "local_pt", "sim_id", "sim_ended", "x_on_cube", "local_min"]
+            gen_specs["persis_in"] = ["x", "x_on_cube", "f", "local_pt", "sim_id", "sim_ended", "local_min"]
```
Comment (jlnav) [inline libensemble/gen_classes/aposmm.py:32]: fix this; an artifact from experimenting with in/out fields

File: libensemble/gen_classes/aposmm.py [hunk 2]
```diff
@@ -28,20 +29,46 @@ def __init__(
...
+            self.tell_buf[field] = results[field]
+
+    @property
+    def _array_size(self):
+        """Output array size must match either initial sample or N points to evaluate in parallel."""
+        user = self.gen_specs["user"]
+        return user["initial_sample_size"] if not self._told_initial_sample else user["max_active_runs"]
+
+    @property
+    def _enough_initial_sample(self):
```
Comment (jlnav) [inline libensemble/gen_classes/aposmm.py:57]: is this necessary for non-optimas cases?

File: libensemble/gen_classes/gpCAM.py [hunk 3]
```diff
@@ -85,7 +85,7 @@ def ask_numpy(self, n_trials: int) -> npt.NDArray:
     def tell_numpy(self, calc_in: npt.NDArray) -> None:
         if calc_in is not None:
             self.y_new = np.atleast_2d(calc_in["f"]).T
-            nan_indices = [i for i, fval in enumerate(self.y_new) if np.isnan(fval)]
+            nan_indices = [i for i, fval in enumerate(self.y_new) if np.isnan(fval[0])]
```
Comment (jlnav) [inline libensemble/gen_classes/gpCAM.py:88]: why was this necessary for gpcam experiments but not libE ones?

File: libensemble/generators.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def final_tell(self, results):
     """
 
     @abstractmethod
-    def __init__(self, *args, **kwargs):
+    def __init__(self, variables: dict[str, List[float]], objectives: dict[str, str], *args, **kwargs):
```
Comment (jlnav) [inline libensemble/generators.py:53]: this change will go in another branch

**Review by shuds13 — APPROVED** [review]
